### ISSUE TITLE
Null pointer dereference

### ISSUE THREAD

**Issue by QiAnXinCodeSafe** [libsdl-org/SDL#11643]
Hi all,
This is Qianxin CodeSafe Team, we found a suspicious issue, allocating memory at
https://github.com/libsdl-org/SDL/blob/1944bc74071451f165adf09b1a79fc1b1dc422a9/src/audio/SDL_audio.c#L237
when we use “SDL_BindAudioStreams_REAL” at 
https://github.com/libsdl-org/SDL/blob/1944bc74071451f165adf09b1a79fc1b1dc422a9/src/audio/SDL_audio.c#L1973
and “SDL_BindAudioStreams_REAL” is uses at 
https://github.com/libsdl-org/SDL/blob/1944bc74071451f165adf09b1a79fc1b1dc422a9/src/audio/SDL_audio.c#L1878
assume the condition  value is true at 
https://github.com/libsdl-org/SDL/blob/1944bc74071451f165adf09b1a79fc1b1dc422a9/src/audio/SDL_audio.c#L1880
and “ObtainLogicalAudioDevice” is uses at 
https://github.com/libsdl-org/SDL/blob/1944bc74071451f165adf09b1a79fc1b1dc422a9/src/audio/SDL_audio.c#L1895
Enter the call from "SDL BindAudioStreams REAL" at
https://github.com/libsdl-org/SDL/blob/1944bc74071451f165adf09b1a79fc1b1dc422a9/src/audio/SDL_audio.c#L385
The loop body executes 0 times at
https://github.com/libsdl-org/SDL/blob/1944bc74071451f165adf09b1a79fc1b1dc422a9/src/audio/SDL_audio.c#L378
assume the condition  value is true at 
https://github.com/libsdl-org/SDL/blob/1944bc74071451f165adf09b1a79fc1b1dc422a9/src/audio/SDL_audio.c#L380
Null pointer stored in 'device'
https://github.com/libsdl-org/SDL/blob/1944bc74071451f165adf09b1a79fc1b1dc422a9/src/audio/SDL_audio.c#L382
The command is returned from ObtainLogicalAudioDevice at
https://github.com/libsdl-org/SDL/blob/1944bc74071451f165adf09b1a79fc1b1dc422a9/src/audio/SDL_audio.c#L1895
Pass a null pointer with argument 1,and Call "UpdateAudioStreamFormatsPhysical" at
https://github.com/libsdl-org/SDL/blob/1944bc74071451f165adf09b1a79fc1b1dc422a9/src/audio/SDL_audio.c#L1964
Enter the call from "SDL BindAudioStreams REAL" at
https://github.com/libsdl-org/SDL/blob/1944bc74071451f165adf09b1a79fc1b1dc422a9/src/audio/SDL_audio.c#L235
Accessing "recording" through the variable "device" results in dereferencing the null pointer at
https://github.com/libsdl-org/SDL/blob/1944bc74071451f165adf09b1a79fc1b1dc422a9/src/audio/SDL_audio.c#L237
in conclusion,it leads to null pointer dereference

**Comment by icculus** [comment]
This is a legitimate bug, thank you for researching this!
